quinn/src/work_limiter.rs

```diff
@@ -102,8 +102,10 @@ impl WorkLimiter {
                 (7.0 * self.smoothed_time_per_work_item_nanos + time_per_work_item_nanos) / 8.0
             };
 
-            self.allowed = ((self.desired_cycle_time.as_nanos()) as f64
-                / self.smoothed_time_per_work_item_nanos) as usize;
+            // Allow at least 1 work item in order to make progress
+            self.allowed = (((self.desired_cycle_time.as_nanos()) as f64
+                / self.smoothed_time_per_work_item_nanos) as usize)
```

Comment:
I wonder if there's also a divide-by-zero risk here, if `now` decides to behave funny (which it absolutely does sometimes).

Reply:
I don't think there really was a risk, because 1ns is really tiny, and I think division by zero is even defined for floating point. But maybe the cast to `usize` not. Anyhow - I now added a protection against that by setting the minimum time to 1ns.

Reply:
> 1ns is really tiny

Right, the issue is that `Instant::now()` isn't necessarily anywhere near that accurate, and moving between cores can do weird things. I believe rust only guarantees monotonicity, and even that only with considerable effort!